analyzers/src/SonarAnalyzer.CSharp/Common/Walkers/CatchLoggingInvocationWalker.cs

```diff
@@ -115,6 +117,22 @@ public override void VisitSimpleLambdaExpression(SimpleLambdaExpressionSyntax no
         // Skip processing to avoid false positives.
     }
 
+    public override void VisitIfStatement(IfStatementSyntax node)
+    {
+        if (visitConditionals)
+        {
+            base.VisitIfStatement(node);
+        }
+    }
+
+    public override void VisitSwitchStatement(SwitchStatementSyntax node)
+    {
+        if (visitConditionals)
+        {
+            base.VisitSwitchStatement(node);
+        }
+    }
```

Comment:
ternaries can throw as well. 
Maybe also override `Conditional` here? (and add a testcase)

![image](https://github.com/SonarSource/sonar-dotnet/assets/115458417/46556027-6b34-4c22-a387-60c44602ca28)


![image](https://github.com/SonarSource/sonar-dotnet/assets/115458417/2be74975-4d3c-445a-9404-739e75ee9749)

Reply:
Forgot about this one. Thanks